### ISSUE TITLE
@Path not setting proper route

### ISSUE THREAD

**Issue by mintu19** [papsign/Ktor-OpenAPI-Generator#35]
I am using @Path like 

```
@Path("/{deviceId}")
data class MyParams(
    @PathParam("Device ID") val deviceId: String
)
```

When i use it with route, proper openapi definitions are created but ktor route is not working. I listed routes and checked... It doesn't show as parameterized route and also none of post method works...

```
route("images") {
                post<MyParams, PResponse, PRequest>(
                    info("Add to server", "Add link")
           ) { pr, req ->
                    respond(pr(true))
          }
```

route listing on ktor show

```
19-04-2020 17:47:04.633 [main] INFO  Application - route: /(method:GET)
19-04-2020 17:47:04.633 [main] INFO  Application - route: /(authenticate "default")/users/agent/(method:GET)
19-04-2020 17:47:04.633 [main] INFO  Application - route: /(authenticate "default")/images/(method:POST)
```

Sending port request to images/1 or images/ routes produces error 404
```
19-04-2020 18:00:01.192 [nioEventLoopGroup-4-3] INFO  Application - Unhandled: POST - /images/
19-04-2020 18:00:32.333 [nioEventLoopGroup-4-3] INFO  Application - Unhandled: POST - /images/1
```

Why route is shown unhandled and it is generated as above without parameters

**Comment by Wicpar** [comment]
On non authenticated routes it does work on the experimental branch.

I find it odd that you declare a post on an authenticated route without giving it the principal type, are you authenticating the route, and if yes why is it not the `OpenAPIAuthenticatedRoute<A>` context that requires you to give the authentication type parameter ?
`fun <reified P : Any, reified R : Any, reified B : Any, A> OpenAPIAuthenticatedRoute<A>.post`

**Comment by mintu19** [comment]
yes i am using authenticated route. I missed writing proper signature while posting here. It is like this: 

`post<MyParams, PResponse, PRequest, IdPrincipal>(`

what is the issue and why route is not properly generated?

**Comment by Wicpar** [comment]
What version are you using ? 0.1 or 0.2 experimental ?

**Comment by mintu19** [comment]
i am using    implementation 'com.github.papsign:Ktor-OpenAPI-Generator:-SNAPSHOT'

**Comment by Wicpar** [comment]
The snapshots might be unstable, can you try the tagged versions to see if
there is one that works ?

Le dim. 19 avr. 2020 à 16:32, Akshit Bhandari <notifications@github.com> a
écrit :

**Comment by mintu19** [comment]
using     implementation 'com.github.papsign:Ktor-OpenAPI-Generator:0.2-beta.2-experimental'

getting:

19-04-2020 21:0```
5:12.678 [main] INFO  Application - route: /openapi.json/(method:GET)
19-04-2020 21:05:12.678 [main] INFO  Application - route: /(method:GET)
19-04-2020 21:05:12.679 [main] INFO  Application - route: /(authenticate "default")/com.papsign.ktor.openapigen.route.util.ConstantRouteSelector@646c0a67/users/agent/(method:GET)
19-04-2020 21:05:12.679 [main] INFO  Application - route: /(authenticate "default")/com.papsign.ktor.openapigen.route.util.ConstantRouteSelector@3804a9a8/images/(method:POST)
```

it doesn't look good

**Comment by Wicpar** [comment]
What is the import you use for the path annotation, is it the one from the library ?

**Comment by mintu19** [comment]
yes

**Comment by mintu19** [comment]
I have verified it is not working without auth also

```
@Path("/my/{id}")
data class TestParam(@PathParam("id") val id: String)
```

```
apiRouting {
        head<TestParam, Boolean>(
            info("id")
        ) {
            respond(true)
        }
    }
```

This will generate 19-04-2020 23:52:15.117 [main] INFO  Application - route: /(method:HEAD)
 instead of route /my/{id}

if i use get instead of head no new route is generated (as get root is already there in normal ktor routes i created).

Path annotation is not working as expected

**Comment by Wicpar** [comment]
You can debug the function `package com.papsign.ktor.openapigen.route.preHandle` at this line:
```kotlin
    val path = P::class.findAnnotation<Path>()
```
this is common code, it should be called by both normal and authenticated routes, it should not be null, and should contain your path string. This is the only relevant code to this issue, if there is a problem it can be seen here...

It could be:
- Annotations are not retained during runtime
- Two annotations with the same name are mixed up
- Two Parameter types are mixed up
- There is an issue that only exists on your system for some reason, this works on mine.

In case it cannot be found you can specify the params in the route directly instead of the annotation: 
```kotlin
route("{param}").post()
```

If you have a code snippet (with all the imports) i can clone that recreates the issue it would be great so i can debug it.

**Comment by mintu19** [comment]
i am not able to debug this via intellij idea. IDE hangs on debug that code...
I will try to make a sample for you to debug...

**Comment by mintu19** [comment]
Using this will generate doc properly?

`route("{param}").post()`

**Comment by Wicpar** [comment]
Yes, the path annotation is just a convenience.

**Comment by mintu19** [comment]
Resolved the issue while creating example project. Route was created properly. It was not displayed in log because ktor have no way to list routes so routes were logged based on selector. I printed all routes and then my routes were visible. Given routes may or may not have handler but there is no way to check as handlers are internal field. So have to list all routes.

Also the api was not called because content Type header was missing while sending. It gets added in postman auto but dont know why it was't. Tracing routes helped with that.

Anyways it works now.Thanks
